tycho-indexer/src/extractor/dynamic_contract_indexer.rs

```diff
@@ -0,0 +1,348 @@
+            self.entrypoint_results
+                .insert(entry_point.clone(), trace.clone());
+        }
+
+        // Get from the storage source (Node) the code, balance and storage changes for the new
+        // results
+        let storage_request: Vec<StorageSnapshotRequest> = traced_entry_points
+            .iter()
+            .flat_map(|traced_entry_point| {
+                traced_entry_point
+                    .tracing_result
+                    .called_addresses
+                    .iter()
+                    .map(|address| {
+                        StorageSnapshotRequest {
+                            address: address.clone(),
+                            // TODO: Set the slots once it's available from the Tracer
+                            slots: None,
+                        }
+                    })
+            })
+            .collect();
+
+        let new_accounts = self
+            .storage_source
+            .get_accounts_at_block(block.into(), &storage_request)
+            .await
+            .expect("Failed to get accounts");
+
+        // Create auxiliary map to prevent looping repeatedely through the data.
+        let mut entrypoint_to_component: HashMap<EntryPointWithData, Vec<ComponentId>> =
+            HashMap::new();
+        for (component_id, entrypoints) in self.protocol_entrypoints.iter() {
+            for entrypoint in entrypoints {
+                entrypoint_to_component
+                    .entry(entrypoint.clone())
+                    .or_insert_with(Vec::new)
+                    .push(component_id.clone());
+            }
+        }
+        for trace in traced_entry_points.iter() {
+            let component_ids = entrypoint_to_component
+                .get(&trace.entry_point)
+                .expect("Failed to get component ID");
+            for account in trace
+                .tracing_result
+                .called_addresses
+                .iter()
+            {
+                if new_accounts.contains(account) {
+                    for component in component_ids {
+                        result_components
+                            .entry(component.clone())
+                            .or_insert_with(Vec::new)
+                            .push(account.clone());
+                    }
+                }
+            }
+        }
+
+        // TODO: Update this to match the specific storage slots. Set (Address, StoreKey) as the key
+        let mut aux_struct: HashMap<Address, Vec<EntryPoint>> = HashMap::new();
+        for (entrypoint_with_data, trace) in self.entrypoint_results {
+            for address in trace.tracing_result.called_addresses {
+                let entrypoint = entrypoint_with_data.entry_point.clone();
+                aux_struct
+                    .entry(address.clone())
+                    .or_insert_with(Vec::new)
+                    .push(entrypoint);
+            }
+        }
```

Comment:
This looks like we should keep this around between blocks? Else we always have a kind of huge memory allocation here on each block?